From 88780c843d1dbd63d18edbeecca17fddfe24a004 Mon Sep 17 00:00:00 2001
From: Fushan Wen <qydwhotmail@gmail.com>
Date: Fri, 18 Aug 2023 22:29:42 +0800
Subject: [PATCH 2/2] TriangleMouseFilter: also check optional position has
 value

If firstEnter is true, only item is set, and interceptedHoverEnterPosition
still has no value until there is another QEvent::HoverEnter event.

Amends 1e8a8d6e6c338bcc66c781863222a038d7136618

CCBUG: 473432


(cherry picked from commit 08c8c00bdd3811b9547360ccb23eeef3da00de54)
---
 components/trianglemousefilter/trianglemousefilter.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/components/trianglemousefilter/trianglemousefilter.h b/components/trianglemousefilter/trianglemousefilter.h
index 086c1d62b..b4d6662ab 100644
--- a/components/trianglemousefilter/trianglemousefilter.h
+++ b/components/trianglemousefilter/trianglemousefilter.h
@@ -29,7 +29,7 @@ struct InterceptedQuickItemData {
 
     explicit operator bool() const
     {
-        return !item.isNull();
+        return !item.isNull() && interceptedHoverEnterPosition.has_value();
     }
 };
 
-- 
2.41.0

